Final/EventDetails.tsx

```diff
@@ -1,27 +1,447 @@
-import { RouteProp, useNavigation, useRoute } from "@react-navigation/native";
-import { Text, View } from "react-native";
+import React, { useEffect, useState } from "react";
+import { View, Text, Image, StyleSheet, Button, ScrollView, Platform, Alert } from "react-native";
 import { SafeAreaView } from "react-native-safe-area-context";
+import { StackNavigationProp } from "@react-navigation/stack";
+import { RouteProp, useNavigation, useRoute } from "@react-navigation/native";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import getAttendingEvents from "./getAttendingEvents";
 import { StackParamList } from "./EventsStack";
-import { NativeStackNavigationProp } from "@react-navigation/native-stack";
-import jsonData from './Events.json';
+import * as calendar from 'expo-calendar';
+import { TouchableOpacity } from "react-native-gesture-handler";
+import Ionicon from "@expo/vector-icons/Ionicons";
+import * as Sharing from 'expo-sharing';
+import Animated from "react-native-reanimated";
+import Events from "./EventsDb";
+import { Share } from "react-native";
+import Contacts from "react-native-contacts";
+import { PushNotification } from "react-native";
+
 
-type ProductDetailNavigationProp = NativeStackNavigationProp<
+interface EventData {
+  id: number;
+  title: string;
+  category: string;
+  date: string;
+  time: string;
+  venue: string;
+  address: string;
+  description: string;
+  image: string;
+}
+
+type EventDetailNavigationProp = StackNavigationProp<
   StackParamList,
   "EventDetails"
 >;
 type EventDetailRouteProp = RouteProp<StackParamList, "EventDetails">;
 
-const eventsLength = jsonData.events.length
 
+const EventDetails: React.FC = () => {
+  const [hasPermission, setHasPermission] = useState<boolean | null>(null);
+  const { params } = useRoute<EventDetailRouteProp>();
+  const navigation = useNavigation<EventDetailNavigationProp>();
+  const [isAttending, setIsAttending] = useState<boolean>(false);
+
+  const event = Events.find((item) => item.id === params.id);
+
+
+  async function requestContactPermissions() {
+    try {
+      await Contacts.requestPermission();
+      console.log("Contact permissions granted.");
+    } catch (error) {
+      console.error("Error requesting contact permissions:", error);
+      Alert.alert("Error", "Failed to request contact permissions.");
+    }
+  }
+
+  useEffect(() => {
+    const getCalendarPermissions = async () => {
```

Comment:
I'd ask for permission when the the "add to calendar" button is pressed instead of the screen renders.